app/forms/field_handlers/duration_handler.py

```diff
@@ -9,7 +9,9 @@ class DurationHandler(FieldHandler):
 
     def get_field(self) -> FormField:
         return FormField(
-            get_duration_form(self.answer_schema, self.error_messages),
+            get_duration_form(
+                self.answer_schema, self.value_source_resolver.schema.error_messages
+            ),
```

Comment:
Not a fan of using attributes from one object for another use case than it was intended for. Normally we make most object attrs private that are needed to the internals of the class only. Doing this makes Value source resolver coupled with the handler even though `field` shouldn't rely on value source.

I personally think it should be changed, but I won't block. I think the reference should be updated to `self.error_messages` which you already have set up in the `__init__`.
```py
self.value_source_resolver = value_source_resolver
self.error_messages = self.value_source_resolver.schema.error_messages or {}
```